### PR TITLE
test: Fix intermittent feature_taproot issue

### PR DESCRIPTION
**Pull request by maflcko**
This fixes three bugs. Also, fix some unrelated code style issues.

Please refer to the commit messages for more information.

### CODE REVIEW

File: test/functional/feature_taproot.py [hunk 1]
```diff
@@ -1436,17 +1437,27 @@ def test_spenders(self, node, spenders, input_counts):
...
-        self.nodes[1].generate(1)
+        rawtx = self.nodes[1].createrawtransaction(
+            inputs=[{
+                'txid': i['txid'],
+                'vout': i['vout']
+            } for i in self.nodes[1].listunspent()],
+            outputs={addr: self.nodes[1].getbalance()},
+        )
+        rawtx = self.nodes[1].signrawtransactionwithwallet(rawtx)['hex']
+        # Transaction is too large to fit into the mempool, so put it into a block
```
Comment (mjdietzx) [inline test/functional/feature_taproot.py:1455]: Where does the error get thrown if you were to send the transaction normally? And why is it too large atm? Everything in the PR looks good, but I thought I should understand this before ack, and haven't been able to figure out

Comment (MarcoFalke) [reply test/functional/feature_taproot.py:1455]: You can find where it is thrown by running `git grep 'Transaction too large'`

Comment (MarcoFalke) [reply test/functional/feature_taproot.py:1455]: The transaction is too large because it has too many inputs

**Comment by laanwj** [comment]
Code review ACK fab900802d74799ac1004265951be1b66e70cf76

**Pull request merged**